src/dialogsearch.cpp

```diff
@@ -82,7 +82,65 @@ void DialogSearch::on_button_search_clicked()
 
     if (ui->radio_variable->isChecked())
     {
-        //TODO implement me
+        int varID = ui->combo_variable->currentData().toInt();
+
+        search_predicate = [varID](const RPG::EventCommand& com)
+        {
+            switch (com.code)
+            {
+                case Cmd::InputNumber:
+                    return com.parameters[1] == varID;
+                case Cmd::ControlSwitches:
+                    return com.parameters[0] == 2 && com.parameters[1] == varID;
+                case Cmd::ControlVars:
+                    return (com.parameters[4] == 1 && com.parameters[5] == varID) ||
+                        (com.parameters[4] == 2 && com.parameters[5] == varID) ||
+                        com.parameters[0] == varID || com.parameters[1] == varID;
```

Comment:
somethings wrong here. This won't display the correct result [http://imgur.com/a/zfLus](http://imgur.com/a/zfLus)